Shopping Site Js/main.py

```diff
@@ -0,0 +1,75 @@
+import json
+import os
+from urllib.parse import urlparse
+import redis
+from jinja2 import Environment
+from jinja2 import FileSystemLoader
+from werkzeug.exceptions import HTTPException
+from werkzeug.exceptions import NotFound
+from werkzeug.middleware.shared_data import SharedDataMiddleware
+from werkzeug.routing import Map
+from werkzeug.routing import Rule
+from werkzeug.urls import url_parse
+from werkzeug.utils import redirect
+from werkzeug.wrappers import Request
+from werkzeug.wrappers import Response
+
+class ShoppingSite(object):
+    def __init__(self, config):
+        self.redis = redis.Redis(config['redis_host'], config['redis_port'])
+        template_path = os.path.join(os.path.dirname(__file__), 'templates')
+        self.jinja_env = Environment(loader=FileSystemLoader(template_path),
+                                 autoescape=True)
+        self.url_map = Map(
+            [
+                Rule("/", endpoint="homepage"),
+            ]
+        )
+    def wsgi_app(self, environ, start_response):
+        request = Request(environ)
+        response = self.dispatch_request(request)
+        return response(environ, start_response)
+
+    def __call__(self, environ, start_response):
+        return self.wsgi_app(environ, start_response)
+    
+    def render_template(self, template_name, **context):
+        t = self.jinja_env.get_template(template_name)
+        return Response(t.render(context), mimetype='text/html')
+    
+    def error_404(self):
+        response = self.render_template("404.html")
+        response.status_code = 404
+        return response
+    
+    def on_homepage(self, request):
+        error = None
+        with open('/home/vivek/Desktop/Shopping Site/item_list.json') as f:
+            data=json.load(f)
+        return self.render_template("homepage.html", error=error, items=data)
```

Comment:
use a relative path instead of an absolute path.
also, it will raise an error when you don't have data.
try to declare an empty list of data.